src/exporters/mod.rs

```diff
@@ -1,5 +1,19 @@
+use std::io::{Result, Write};
+
+use crate::{Operation, WireValue};
+
 mod bristol;
 mod json;
 
-pub use bristol::bool_circuit_to_bristol;
+pub use bristol::BristolFashion;
 pub use json::bool_circuit_to_json;
+
+/// The core export trait.
+///
+/// Individual exporters (such as for Bristol-fashion circuits) are expected
+/// to implement this trait.
+pub trait Export<T: WireValue> {
+    fn export_gate(gate: &Operation<T>, sink: &mut impl Write) -> Result<()>;
+
+    fn export_circuit(gates: &[Operation<T>], witness: &[T], sink: &mut impl Write) -> Result<()>;
+}
```

Comment:
I'm not sure if the generic `T: WireValue` bound here is the right abstraction, but it's better than what we had before.